addons/website_livechat/models/website.py

```diff
@@ -18,15 +17,16 @@ class website_config_settings(models.TransientModel):
 
 
 
-class ir_ui_view(osv.Model):
+class IrUiView(models.Model):
 
     _inherit = "ir.ui.view"
 
-    def _prepare_qcontext(self, cr, uid, context=None):
-        qcontext = super(ir_ui_view, self)._prepare_qcontext(cr, uid, context=context)
+    @api.model
+    def _prepare_qcontext(self):
+        qcontext = super(IrUiView, self)._prepare_qcontext()
         if request and getattr(request, 'website_enabled', False):
             if request.website.channel_id:
-                qcontext['website_livechat_url'] = self.pool.get('ir.config_parameter').get_param(cr, SUPERUSER_ID, 'web.base.url')
-                qcontext['website_livechat_dbname'] = cr.dbname
+                qcontext['website_livechat_url'] = self.env['ir.config_parameter'].get_param('web.base.url')
+                qcontext['website_livechat_dbname'] = self._cr.dbname
```

Comment:
`Model._cr` should be avoided, they're here for BC reasons. Get the cursor from the env.